File: LAB2/Line.cpp
#include "Line.h"
#include<iostream>
using namespace std;

Line::Line()
{
    size = 100;
    str = new char[size];

}
int  Line::maxwlen()
{
    return strlen(str);
}

void  Line::inputString()
{
    cin.getline(str, size);
}

void  Line::printString()
{
    cout << str << endl;
}

void  Line::lineupper()
{
    for (int i = 0; i < strlen(str); i++)
    {
        if ((str[i - 1] == ' ') || (i == 0))
            str[i] = toupper(str[i]);
    }
}
bool Line::search(char ch)
{
    string s = str;
    int i = 0;
    bool f = false;
    for (i = s.find(ch, i++); i != string::npos; i = s.find(ch, i + 1))
        f = true;
    return f;
}

File: LAB2/Line.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LAB2_OOP_CS
{
    class Line
    {
        private char[] word;
        private int size;

        public char[] str;

        public Line()
        {
            size = 100;
            str = new char[size];

        }

        public int maxwlen()
        {
            return str.Length;
        }

        public void inputString()
        {
            str = Console.ReadLine().ToCharArray();
        }

        public void printString()
        {
            Console.WriteLine(str);
        }

        public void lineupper()
        {
            for (int i = 0; i < str.Length; i++)
            {
                if ((str[i] == ' ') || (i == 0))
                    str[i] = char.ToUpper(str[i]);
            }
        }

        public bool search(char ch)
        {
            string s = new string(str);
            bool f = false;
            if (s.Contains(ch.ToString()))
                f = true;
            return f;
        }
    }
}
